File: Medium/2938. Separate Black and White Balls.java
class Solution {
    public long minimumSteps(String s) {
        int upperBound = s.length() - 1;
        for (int i = s.length() - 1; i > -1; i--) {
            if (s.charAt(i) == '0') {
                break;
            }
            upperBound--;
        }
        long totalSteps = 0L;
        for (int i = upperBound; i > -1; i--) {
            if (s.charAt(i) == '0') {
                continue;
            }
            totalSteps += (upperBound - i);
            upperBound--;
        }
        return totalSteps;
    }
}

File: Medium/2938. Separate Black and White Balls.py
class Solution:
    def minimumSteps(self, s: str) -> int:
        n = len(s)
        upper_bound = n - 1
        while (upper_bound >= 0 and s[upper_bound] == '1'):
            upper_bound -= 1
        res = 0
        for i in range(upper_bound, -1, -1):
            if s[i] == '1':
                res += upper_bound - i
                upper_bound -= 1
        return res